### PR TITLE
[Enhancement] avoid duplicate deletion of schema

### PR DESCRIPTION
**Pull request by wuzhanpeng**
### Motivation

Currently when I need to delete a namespace forcedly, the `NamespacesBase#internalDeleteNamespaceForcefully` will firstly list all the topics under the namespace and then delete them concurrently without distinguishing wheather the topic is partitioned or non-partitioned. This behaivor will cause duplicate deletion of schema when the topic list contains partitioned topics, because all the partitions of the partitioned topic share the same schema.

### Modifications

Distinguish partitioned topic to avoid duplicate deletion of the same schema.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): no
  - The public API: no
  - The schema: no
  - The default values of configurations: no
  - The wire protocol: no
  - The rest endpoints: no
  - The admin cli options: no
  - Anything that affects deployment: no

### Documentation

This PR does not need to update the docs.

### CODE REVIEW

**Comment by eolivelli** [comment]
I believe that this is a good idea.

Can you please add a test case ?

**Comment by Anonymitaet** [comment]
@hangc0276 thanks for triaging PR and issues. When you label PRs, if you find some PRs do not need to update doc,  could you please help label the PR with `no-need-doc` in the future? Thanks

**Comment by hangc0276** [comment]
> @hangc0276 thanks for triaging PR and issues. When you label PRs, if you find some PRs do not need to update doc, could you please help label the PR with `no-need-doc` in the future? Thanks

@Anonymitaet Ok,  I will add doc label in the following PRs, thank you!

**Review by eolivelli — CHANGES_REQUESTED** [review]
Good work 
Can you add a test case?

**Comment by wuzhanpeng** [comment]
> Good work
> Can you add a test case?

Sorry for the late reply! I will add a test later.

**Comment by wuzhanpeng** [comment]
@eolivelli @hangc0276 Sorry for the late. I supplemented a test case. PTAL~

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiTest2.java [hunk 1]
```diff
@@ -1472,6 +1474,60 @@ public void testForceDeleteNamespace() throws Exception {
...
+        final String np = "persistent://" + ns + "/np";
+
+        admin.topics().createPartitionedTopic(p1, 1);
+        admin.topics().createPartitionedTopic(p5, 5);
+        admin.topics().createNonPartitionedTopic(np);
+
+        final String exNp = "persistent://" + exNs + "/np";
+        admin.topics().createNonPartitionedTopic(exNp);
+        // insert an invalid topic name
+        pulsar.getLocalMetadataStore().put(
```
Comment (eolivelli) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiTest2.java:1496]: why this is related to this patch ?

Comment (wuzhanpeng) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminApiTest2.java:1496]: There are actually three scenarios I tested for partitioned topics, non-partitioned topics and exceptions while handling internal deletion.

**Review by eolivelli — APPROVED** [review]
LGTM

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java [hunk 2]
```diff
@@ -415,14 +415,40 @@ protected void internalDeleteNamespaceForcefully(AsyncResponse asyncResponse, bo
...
+                                        + "and non-partitioned-topics:%s have been sent out asynchronously. "
+                                        + "Reason: %s",
+                                topic, partitionedTopics, nonPartitionedTopics, e.getCause());
+                        log.error("[{}] {}", clientAppId(), errorMessage, e);
+                        asyncResponse.resume(new RestException(Status.INTERNAL_SERVER_ERROR, errorMessage));
+                        return;
                     }
                 }
+
+                log.info("Successfully send deletion command of partitioned-topics:{} "
```
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java:449]: It should be a debug level log? If we have many partitioned topics and non-partitioned-topics, we will get a very long log here, maybe print the topic count is better?

Comment (wuzhanpeng) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java]: Thanks for your suggestion! I have addressed the comment. PTAL~ @codelipenghui

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**